src/ctap2.rs

```diff
@@ -217,13 +217,13 @@ pub type SerializedAuthenticatorData = Bytes<AUTHENTICATOR_DATA_LENGTH>;
 
 // The reason for this non-use of CBOR is for compatibility with
 // FIDO U2F authentication signatures.
-impl<A: SerializeAttestedCredentialData, E: serde::Serialize> AuthenticatorData<A, E> {
+impl<'a, A: SerializeAttestedCredentialData, E: serde::Serialize> AuthenticatorData<'a, A, E> {
     #[inline(never)]
-    pub fn serialize(&self) -> SerializedAuthenticatorData {
+    pub fn serialize(&self) -> Result<SerializedAuthenticatorData> {
```

Comment:
Since this now returns a `Result`, why not  remove the `.unwrap()`s here like in `AttestedCredentialData` ?

Reply:
Right, done.